### ISSUE TITLE
Possible to reduce the NPM warnings upon install?

### ISSUE THREAD

**Issue by Frijol** [tessel/t2-cli#854]
The install works, but it looks .. bad. I've had people who are new to Node become very nervous to see this number of warnings, and expect that the install has failed.

Here's what I get when I install our CLI globally:

```sh
➜  t2-cli git:(master) npm i -g t2-cli
npm WARN engine mdns-js@0.4.0: wanted: {"node":"^0.10.3 || ^0.12.0"} (current: {"node":"4.2.3","npm":"2.14.7"})
\
> usb@1.1.2 install /usr/local/lib/node_modules/t2-cli/node_modules/usb
> node-pre-gyp install --fallback-to-build

[usb] Success: "/usr/local/lib/node_modules/t2-cli/node_modules/usb/src/binding/usb_bindings.node" is installed via remote
/usr/local/bin/t2 -> /usr/local/lib/node_modules/t2-cli/bin/tessel-2.js

> t2-cli@0.0.27 postinstall /usr/local/lib/node_modules/t2-cli
> t2 install-drivers --loglevel=error || true; t2 install-homedir --loglevel=error || true;

npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/npmconf requires semver@'2' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/semver,
npm WARN unmet dependency which is version 5.0.3
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/npmconf requires uid-number@'0.0.5' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/uid-number,
npm WARN unmet dependency which is version 0.0.6
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/read-package-json/node_modules/normalize-package-data requires semver@'2 || 3' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/semver,
npm WARN unmet dependency which is version 5.0.3
npm WARN unmet dependency /usr/local/lib/node_modules/tumblr.js/node_modules/request/node_modules/http-signature/node_modules/sshpk requires assert-plus@'^1.0.0' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/tumblr.js/node_modules/request/node_modules/http-signature/node_modules/assert-plus,
npm WARN unmet dependency which is version 0.2.0
t2-cli@0.0.27 /usr/local/lib/node_modules/t2-cli
├── is-root@1.0.0
├── bindings@1.2.1
├── url-join@0.0.1
├── char-spinner@1.0.1
├── shell-escape@0.2.0
├── semver@5.3.0
├── progress@1.1.8
├── async@1.5.2
├── common-tags@0.0.1
├── colors@1.1.2
├── stream-to-buffer@0.1.0 (stream-to@0.2.2)
├── fstream-ignore@1.0.5 (inherits@2.0.1)
├── toml@2.3.0
├── osenv@0.1.3 (os-homedir@1.0.1, os-tmpdir@1.0.1)
├── minimatch@3.0.3 (brace-expansion@1.1.6)
├── usb-daemon-parser@0.0.2 (debug@2.2.0)
├── acorn@3.3.0
├── glob@5.0.15 (path-is-absolute@1.0.0, inherits@2.0.1, inflight@1.0.5, once@1.3.3)
├── tar@2.2.1 (block-stream@0.0.9, inherits@2.0.1)
├── nomnom@1.8.1 (underscore@1.6.0, chalk@0.4.0)
├── promzard@0.3.0 (read@1.0.7)
├── sshpk@1.10.0 (assert-plus@1.0.0, dashdash@1.14.0, getpass@0.1.6, jsbn@0.1.0, ecc-jsbn@0.1.1, tweetnacl@0.13.3, asn1@0.2.3, jodid25519@1.0.2, bcrypt-pbkdf@1.0.0)
├── fs-temp@1.1.1 (random-path@0.1.0)
├── tar-stream@1.5.2 (xtend@4.0.1, end-of-stream@1.1.0, readable-stream@2.1.5, bl@1.1.2)
├── fstream@1.0.10 (inherits@2.0.1, graceful-fs@4.1.6, mkdirp@0.5.1, rimraf@2.5.4)
├── ssh2@0.4.15 (readable-stream@1.0.34, ssh2-streams@0.0.23)
├── fs-extra@0.26.7 (path-is-absolute@1.0.0, jsonfile@2.3.1, graceful-fs@4.1.6, klaw@1.3.0, rimraf@2.5.4)
├── npmlog@4.0.0 (set-blocking@2.0.0, console-control-strings@1.1.0, are-we-there-yet@1.1.2, gauge@2.6.0)
├── update-notifier@0.6.3 (is-npm@1.0.0, semver-diff@2.1.0, chalk@1.1.3, boxen@0.3.1, configstore@2.1.0, latest-version@2.0.0)
├── mdns-js@0.4.0 (semver@4.3.6, debug@2.2.0, mdns-js-packet@0.1.12)
├── uglify-js@2.7.3 (async@0.2.10, uglify-to-browserify@1.0.2, source-map@0.5.6, yargs@3.10.0)
├── t2-project@0.4.0 (builtins@1.0.3, glob@7.0.6, resolve@1.1.7, array-includes@3.0.2, module-deps@4.0.7)
├── node-rsa@0.2.30 (asn1@0.2.3, lodash@3.3.0)
├── inquirer@0.8.5 (ansi-regex@1.1.1, through@2.3.8, cli-width@1.1.1, figures@1.7.0, readline2@0.1.1, chalk@1.1.3, lodash@3.10.1, rx@2.5.3)
├── request@2.74.0 (tunnel-agent@0.4.3, aws-sign2@0.6.0, oauth-sign@0.8.2, forever-agent@0.6.1, is-typedarray@1.0.0, caseless@0.11.0, stringstream@0.0.5, aws4@1.4.1, isstream@0.1.2, json-stringify-safe@5.0.1, extend@3.0.0, tough-cookie@2.3.1, qs@6.2.1, node-uuid@1.4.7, mime-types@2.1.11, combined-stream@1.0.5, bl@1.1.2, hawk@3.1.3, http-signature@1.1.1, har-validator@2.0.6, form-data@1.0.1)
├── usb@1.1.2 (nan@2.4.0)
└── npm@2.15.10
```

(cc @tcr)

**Comment by LinusU** [comment]
> npm WARN engine mdns-js@0.4.0

It seems like just upgrading to mdns-js 0.5.0 would fix this. The only changes seems to be dropping Node 0.x support.

**Comment by Frijol** [comment]
When you say "fresh install", do you mean of Node or of t2-cli?

I'm on Node 4.2.3, trying update of Node to 4.5.0

**Comment by LinusU** [comment]
I meant fresh of t2-cli, as in `npm rm -g t2-cli && npm i -g t2-cli` :)

**Comment by Frijol** [comment]
Sounds good, once I deleted my tumblr.js module install this worked for me too

**Comment by LinusU** [comment]
Cool, the first warning is fixed by #855
